polkadot/node/core/provisioner/src/lib.rs

```diff
@@ -163,23 +220,74 @@ async fn run_iteration<Context>(
 	}
 }
 
-async fn handle_active_leaves_update(
+#[overseer::contextbounds(Provisioner, prefix = self::overseer)]
+async fn handle_active_leaves_update<Context>(
+	ctx: &mut Context,
 	update: ActiveLeavesUpdate,
 	per_relay_parent: &mut HashMap<Hash, PerRelayParent>,
 	inherent_delays: &mut InherentDelays,
+	slot_delays: &mut SlotDelays,
+	inherents: &mut LruMap<Hash, ProvisionerInherentData>,
+	metrics: &Metrics,
 ) -> Result<(), Error> {
 	gum::trace!(target: LOG_TARGET, "Handle ActiveLeavesUpdate");
 	for deactivated in &update.deactivated {
 		per_relay_parent.remove(deactivated);
 	}
 
-	if let Some(leaf) = update.activated {
-		gum::trace!(target: LOG_TARGET, leaf_hash=?leaf.hash, "Adding delay");
-		let delay_fut = Delay::new(PRE_PROPOSE_TIMEOUT).map(move |_| leaf.hash).boxed();
-		per_relay_parent.insert(leaf.hash, PerRelayParent::new(leaf));
-		inherent_delays.push(delay_fut);
-	}
+	let Some(leaf) = update.activated else { return Ok(()) };
+
+	gum::trace!(target: LOG_TARGET, leaf_hash=?leaf.hash, "Adding delay");
+	let delay_fut = Delay::new(PRE_PROPOSE_TIMEOUT).map(move |_| leaf.hash).boxed();
+	per_relay_parent.insert(leaf.hash, PerRelayParent::new(leaf.clone()));
+	inherent_delays.push(delay_fut);
+
+	let slot_delay = time_until_next_slot(Duration::from_millis(6000));
```

Comment:
Wouldn't this actually give you a 6s delay from current leaf + 2s extra ? Hence will end up calling send_inherent_data_bg a bit late for this hash.